atc/db/migration/migrations/1548261635_create_table_resource_config_scopes.down.sql

```diff
@@ -0,0 +1,23 @@
+BEGIN;
+
+  ALTER TABLE resources
+    DROP COLUMN resource_config_scope_id;
+
+  ALTER TABLE resource_config_versions
+    DROP COLUMN resource_config_scope_id,
+    ADD COLUMN "resource_config_id" integer NOT NULL REFERENCES resource_configs (id) ON DELETE CASCADE,
```

Comment:
Upgrading to this with existing data fails because of this new column:

```
web_1     | failed to migrate database: 1 error occurred:
web_1     | 	* Migration '1548261635_create_table_resource_config_scopes.up.sql' failed: 2 errors occurred:
web_1     | 	* Transaction ALTER TABLE resource_config_versions
web_1     |     DROP COLUMN resource_config_id,
web_1     |     ADD COLUMN resource_config_scope_id integer NOT NULL REFERENCES resource_config_scopes (id) ON DELETE CASCADE,
web_1     |     ADD CONSTRAINT "resource_config_scope_id_and_version_md5_unique" UNIQUE ("resource_config_scope_id", "version_md5") failed, rolled back the migration
web_1     | 	* pq: column "resource_config_scope_id" contains null values
web_1     | 
web_1     | 
web_1     | 
web_1     | 
```

Looks like we need to deal with existing rows in `resource_config_versions`.

This only affects users upgrading from `master` since that whole table is new as of 5.0, so it should be empty when the migrations run. But maybe we should make the migration more resilient anyway, if not just for prod's sake.

What should we do here? Just truncate the table?

Reply:
I think as much resiliency we can build into migrations is always good. If we ever wanted to release nightly stable version of Concourse it would be helpful, and in general migrations should be considered separately from what version they're in if only to maintain our own sanity when shipping and recommending upgrade paths.

It seems like `resource_config_versions` is a "meta" table with data generated by the scheduler, so truncating doesn't really lose anything Concourse can't regenerate later